Disallow unquoted database names with dot (#1948)
neo4j/docs-operations: modules/ROOT/pages/database-administration/standard-databases/naming-databases.adoc

The `-` (dash) and `.` (dot) characters are not legal in Cypher variables.
Names with a `-` in them must be enclosed within backticks.
For example, `CREATE DATABASE ++`main-db`++` is a valid database name.

> Since Neo4j 5.27 names with a `.` in them must be enclosed within backticks.

Author: ```suggestion
Since Cypher 25 names with a `.` in them must be enclosed within backticks.
```

> However, database names with dots in them are not recommended, due to the difficulty of determining if a dot is part of the database name or a delimiter for a database alias in a composite database.

Reviewer: ```suggestion
However, using dots in database names is not recommended, as it makes it difficult to determine if a dot is part of the database name or a delimiter for a database alias in a composite database.
```

> A future version of Neo4j may disallow database names with dots entirely.

Reviewer: ```suggestion
A future version of Neo4j may entirely disallow database names with dots.
```
